File: static.ml

let mean lst =
  let total = List.fold_left (+) 0 lst in
  float_of_int total /. float_of_int (List.length lst)

let median lst =
  let sorted = List.sort compare lst in
  let len = List.length sorted in
  if len mod 2 = 0 then
    let mid1 = List.nth sorted (len / 2 - 1) in
    let mid2 = List.nth sorted (len / 2) in
    float_of_int (mid1 + mid2) /. 2.0
  else
    float_of_int (List.nth sorted (len / 2))

let mode lst =
  let count_map =
    List.fold_left (fun acc x ->
      if List.mem_assoc x acc then
        (x, List.assoc x acc + 1) :: List.remove_assoc x acc
      else
        (x, 1) :: acc
    ) [] lst in
  let max_freq = List.fold_left (fun m (_, c) -> max m c) 0 count_map in
  List.fold_right (fun (k, v) acc -> if v = max_freq then k :: acc else acc) count_map []

let () =
  let data = [10; 20; 20; 30; 40; 40; 40; 50] in
  Printf.printf "Mean: %.2f\n" (mean data);
  Printf.printf "Median: %.2f\n" (median data);
  Printf.printf "Mode: ";
  List.iter (fun x -> Printf.printf "%d " x) (mode data);
  print_newline ()

File: static.py
from collections import Counter

class StatisticsCalculator:
    def __init__(self, data):
        self.data = sorted(data)

    def mean(self):
        return sum(self.data) / len(self.data)

    def median(self):
        n = len(self.data)
        mid = n // 2
        if n % 2 == 0:
            return (self.data[mid - 1] + self.data[mid]) / 2
        else:
            return self.data[mid]

    def mode(self):
        freq = Counter(self.data)
        max_freq = max(freq.values())
        return [val for val, count in freq.items() if count == max_freq]

if __name__ == "__main__":
    values = [10, 20, 20, 30, 40, 40, 40, 50]
    stats = StatisticsCalculator(values)

    print("Mean:", round(stats.mean(), 2))
    print("Median:", round(stats.median(), 2))
    print("Mode:", stats.mode())